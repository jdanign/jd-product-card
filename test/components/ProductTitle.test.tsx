import * as React from 'react';
import renderer from 'react-test-renderer';
import { ProductCard, ProductTitle } from "../../src/components";
import { products } from '../data/products';


describe('ProductTitle', () => { 
    test('Debe mostrar el componente correctamente con el tÃ­tulo personalizado', () => { 
        const wrapper = renderer.create(
            <ProductTitle title='Custom Product' className='custom-class' />
        )

        expect(wrapper.toJSON()).toMatchSnapshot();
    })

    test('debe mostrar el componente con el nombre del producto', () => {
        const wrapper = renderer.create(
            <ProductCard product={products[0]} className='custom-class'>
                {
                    ()=>(
                        <ProductTitle />
                    )
                }
            </ProductCard>
        )

        expect(wrapper.toJSON()).toMatchSnapshot();
    })
})