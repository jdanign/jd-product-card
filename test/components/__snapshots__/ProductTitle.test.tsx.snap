// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductTitle Debe mostrar el componente correctamente con el t√≠tulo personalizado 1`] = `
<span
  className="productDescription custom-class"
>
  Custom Product
</span>
`;

exports[`ProductTitle debe mostrar el componente con el nombre del producto 1`] = `
<div
  className="productCard custom-class"
>
  <span
    className="productDescription "
  >
    Coffee Mug - Card
  </span>
</div>
`;
